### ISSUE TITLE
Authentication passes with an incorrect nonce

### ISSUE THREAD

**Issue by samuel-lucas6** [soatok/experimental-caead#1]
I may be wrong, but it looks like authentication can pass when the first 160 bits of the nonce are correct but the last 96 bits are not because that latter part of the nonce is not in the key derivation or authentication tag calculation.

A solution would be to include the entire nonce as part of the key derivation. I'm currently working on an [Internet Draft](https://github.com/samuel-lucas6/draft-lucas-generalised-committing-aead) that discusses that.

**Comment by soatok** [comment]
Yep. This was thrown together to demonstrate a concept. I didn't take the time to carefully analyze the security of this construct.
